internal/cli/version.go

```diff
@@ -0,0 +1,185 @@
+`
+
+// UpdateVersionMsgFormat is the message displayed to a user when an older
+// version of the CLI is installed.
+const UpdateVersionMsgFormat string = `
+  We need to update your New Relic CLI version to continue.
+
+    Installed version: %s
+    Latest version:    %s
+
+  To update your CLI and continue this installation, run this command:
+
+    %s
+`
+
+// Version returns the version of the CLI that's currently installed.
+func Version() string {
+	if version == "" {
+		version = os.Getenv("NEW_RELIC_CLI_VERSION")
+	}
+
+	return version
+}
+
+// IsLatestVersion returns true if the provided version string matches
+// the current installed version.
+func IsLatestVersion(ctx context.Context, latestVersion string) (bool, error) {
+	installedVersion, err := semver.NewVersion(Version())
+	if err != nil {
+		return false, fmt.Errorf("error parsing installed CLI version %s: %s", version, err.Error())
+	}
+
+	lv, err := semver.NewVersion(latestVersion)
+	if err != nil {
+		return false, fmt.Errorf("error parsing version to check %s: %s", latestVersion, err.Error())
+	}
+
+	if installedVersion.LessThan(lv) {
+		return false, nil
+	}
+
+	return installedVersion.Equal(lv), nil
+}
+
+// GetLatestReleaseVersion returns the latest released tag.
+// The latest tag is pulled from the newrelic-cli GitHub repository.
+func GetLatestReleaseVersion(ctx context.Context) (string, error) {
+	if latestVersion != "" {
+		return latestVersion, nil
+	}
+
+	latestRelease, err := fetchLatestRelease(ctx)
+	if err != nil {
+		return "", fmt.Errorf("error fetching latest release: %s", err.Error())
+	}
+
+	lv, err := semver.NewVersion(latestRelease.TagName)
+	if err != nil {
+		return "", fmt.Errorf("error parsing latest release version string '%s': %s", latestRelease.TagName, err.Error())
+	}
+
+	// Cache the latest version string
+	versionString := lv.String()
+	setLatestVersion(versionString)
+
+	return versionString, nil
+}
+
+// IsDevEnvironment is a naive implementation to determine if the CLI
+// is being run in a dev environment. IsDevEnvironment returns true when
+// the installed CLI version is either in a prerelease state or in a dirty state.
+// The version string is generated at compile time using git. The prerelease string
+// is appended to the primary semver version string.
+//
+// If you're doing local development on the CLI, your version may look similar to
+// the examples below.
+//
+// Examples of versions that have a prerelease tag (i.e. the suffix):
+//
+//  v0.32.1-10-gbe63a24
+//  v0.32.1-10-gbe63a24-dirty
+//
+// In this example version string, "10" represents the number of commits
+// since the 0.32.1 tag was created. The "gbe63a24" is the previous commit's
+// abbreviated sha. The "dirty" part means that git was in a dirty state at compile
+// time, meaning an updated file was saved, but not yet committed.
+func IsDevEnvironment() bool {
+	v, err := semver.NewVersion(version)
+	if err != nil {
+		return true
```

Comment:
Does it make sense to return true here if we have a nonsensical version string?  Would it make more sense to return the error?

Reply:
Fair question and open to suggestions. This was a weird decision to make. The only time I've really seen a nonsensical version string is during development, but I guess there's still a chance of us actually releasing a really weird tag, although it's highly unlikely and would likely get rectified soon after.